rocketpool/watchtower/submit-rpl-price.go

```diff
@@ -367,15 +369,46 @@ func (t *submitRplPrice) run(state *state.NetworkState) error {
 	lastSubmissionBlock := state.NetworkDetails.PricesBlock
 
 	referenceTimestamp := t.cfg.Smartnode.PriceBalanceSubmissionReferenceTimestamp.Value.(int64)
+
 	// Get the duration in seconds for the interval between submissions
 	submissionIntervalInSeconds := int64(state.NetworkDetails.PricesSubmissionFrequency)
 	eth2Config := state.BeaconConfig
 
-	_, nextSubmissionTime, targetBlockHeader, err := utils.FindNextSubmissionTarget(t.rp, eth2Config, t.bc, t.ec, lastSubmissionBlock, referenceTimestamp, submissionIntervalInSeconds)
-	if err != nil {
-		return err
+	var hasSubmittedPastBlock bool
+	var nextSubmissionTime time.Time
+	var targetBlockNumber uint64
+	var lastSubmissionBlockHeader *types.Header
+
+	// Check if the node has submitted prices for the latest block
+	if lastSubmissionBlock != 0 {
+		// Get the lastSubmissionBlock timestamp
+		lastSubmissionBlockHeader, err = t.rp.Client.HeaderByNumber(context.Background(), big.NewInt(int64(lastSubmissionBlock)))
+		if err != nil {
+			t.log.Printlnf("Error getting the latest submission block header: %s", err.Error())
+			return err
+		}
+		hasSubmittedPastBlock, err = t.hasSubmittedSpecificBlockPrices(nodeAccount.Address, lastSubmissionBlock, lastSubmissionBlockHeader.Time, state.NetworkDetails.RplPrice)
+		if err != nil {
+			t.log.Printlnf("Error checking if node has submitted prices for block %d: %s", lastSubmissionBlock, err.Error())
+			return err
+		}
+	}
+
+	if !hasSubmittedPastBlock {
```

Comment:
you can move the `else` part of this conditional to the above conditional, because it only runs if `hasSubmittedSpecificBlockPrices` returns true there. Then you can change this conditional to `if targetBlockNumber == 0`, which is a bit more semantic